courses/Ada_For_The_C_Embedded_Developer/chapters/01_Introduction.rst

```diff
@@ -0,0 +1,116 @@
+:prev_state: False
+
+:code-config:`run_button=True;prove_button=False;accumulate_code=False`
+
+:code-config:`reset_accumulator=True`
+
+.. role:: ada(code)
+   :language: ada
+
+.. role:: c(code)
+   :language: c
+
+Introduction
+==============
+
+So, what is this Ada thing anyway?
+-----------------------------------
+
+To answer this question let’s introduce Ada as it compares to C for an embedded application. C developers are used to a certain coding semantic and style of programming. Especially in the embedded domain, developers are used to working at a very low level near the hardware to directly manipulate memory and registers. Normal operations involve mathematical operations on pointers, complex bit shifts, and logical bitwise operations. C is well designed for such operations as it is a low level language that was designed to replace assembly language for faster, more efficient programming. Because of this minimal abstraction, the programmer has to model the data that represents the problem they are trying to solve using the language of the physical hardware.
+
+Let’s look at an example of this problem in action by comparing the same program in Ada and C:
+
+.. code:: c cli_input
+
+   !main.c
+   #include <stdio.h>
+   #include <stdlib.h>
+
+   #define DEGREES_MAX           (360)
+   typedef unsigned int degrees;
+
+   #define MOD_DEGREES(x)   (x % DEGREES_MAX)
+
+   degrees add_angles(degrees* list, int length)
+   {
+     degrees sum = 0;
+     for(int i = 0; i < length; ++i) {
```

Comment:
Minor comment: I'd prefer a coding style where you use a whitespace after C-keywords (e.g.: ``for (...)``, ``if (...)``, etc).

Reply:
Sure thats fine with me.